GitUI/SpellChecker/WordAtCursorExtractor.cs

```diff
@@ -1,41 +1,83 @@
-﻿using System.Text;
+﻿using System;
+using System.Text;
 
 namespace GitUI.SpellChecker
 {
     internal class WordAtCursorExtractor : IWordAtCursorExtractor
     {
         public string Extract(string text, int cursor)
         {
+            var startIndex = FindStartOfWord(text, cursor);
+            return startIndex < 0 ? string.Empty : text.Substring(startIndex, cursor - startIndex + 1);
+        }
+
+        public (int start, int length) FindWord(string text, int cursor)
+        {
+            int start = Math.Min(FindStartOfWord(text, cursor), cursor);
+            if (start < 0)
+            {
+                return (-1, 0);
+            }
+
+            int end = FindEndOfWord(text, cursor);
+            return (start, Math.Max(1, end - start));
+        }
+
+        public int FindStartOfWord(string text, int cursor)
```

Comment:
I prefer to expose only members that are required. 
Unless there is a compelling reason to expose members to the outside, please consider making them `internal` (for tests access).